src/visualizations/config/generators/dhis/singleValue.js

```diff
@@ -265,17 +290,59 @@ const generateDVItem = (config, { legendSet, parentEl, fontStyle, noData }) => {
     return svg
 }
 
+const shouldUseContrastColor = (inputColor) => {
+    // based on https://stackoverflow.com/questions/3942878/how-to-decide-font-color-in-white-or-black-depending-on-background-color
+    var color =
+        inputColor.charAt(0) === '#' ? inputColor.substring(1, 7) : inputColor
+    var r = parseInt(color.substring(0, 2), 16) // hexToR
+    var g = parseInt(color.substring(2, 4), 16) // hexToG
+    var b = parseInt(color.substring(4, 6), 16) // hexToB
+    var uicolors = [r / 255, g / 255, b / 255]
+    var c = uicolors.map((col) => {
+        if (col <= 0.03928) {
+            return col / 12.92
+        }
+        return Math.pow((col + 0.055) / 1.055, 2.4)
+    })
+    var L = 0.2126 * c[0] + 0.7152 * c[1] + 0.0722 * c[2]
+    return L <= 0.179
+}
+
 export default function (
     config,
     parentEl,
-    { dashboard, legendSets, fontStyle, noData }
+    { dashboard, legendSets, fontStyle, noData, legendOptions }
 ) {
-    const legendSet = legendSets[0]
+    const legendSet = legendOptions && legendSets[0]
+    const legendColor =
+        legendSet && getColorByValueFromLegendSet(legendSet, config.value)
+    let valueColor, titleColor
+    if (legendColor) {
+        if (legendOptions.style === LEGEND_DISPLAY_STYLE_FILL) {
+            parentEl.style.background = legendColor
```

Comment:
Set `backgroundColor`? Very minor.

Reply:
The `background` shorthand works just as well and you might even argue that it's technically more correct (since it resets all the other background properties, which in this case is what we want). There doesn't seem to be a best practice for this though (we use both props in our repos).